game/src/main/kotlin/org/apollo/game/plugin/kotlin/KotlinPluginScript.kt

```diff
@@ -61,3 +66,22 @@ class KotlinMessageHandler<T : Message>(val world: World, val context: PluginCon
     }
 
 }
+
+class KotlinCommandListener(val level: PrivilegeLevel, val function: (Player, Command) -> Unit) : CommandListener(level) {
+
+    override fun execute(player: Player, command: Command) {
+        function.invoke(player, command)
+    }
+
+}
+
+class KotlinCommandHandler(val world : World, val command: String, val privileges: PrivilegeLevel) {
+
+    var function: (Player, Command) -> Unit = { _, _ -> }
+
+    fun then(function: (Player, Command) -> Unit) {
```

Comment:
If this is changed to `Command.(Player) -> Unit` then it'll be invoked from the `Command`'s context (just like `.then` in the message handlers) and you'll only have the lone `Player` argument.